test/jdk/sun/security/ssl/X509KeyManager/NullCases.java

```diff
@@ -32,50 +32,173 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.X509KeyManager;
+import java.security.KeyPair;
+import java.security.SecureRandom;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
 import java.security.cert.X509Certificate;
 
-import java.util.*;
+import static jdk.httpclient.test.lib.common.DynamicKeyStoreUtil.generateCert;
+import static jdk.httpclient.test.lib.common.DynamicKeyStoreUtil.generateKeyStore;
+import static jdk.httpclient.test.lib.common.DynamicKeyStoreUtil.generateRSAKeyPair;
 
 
 public class NullCases {
-    public static void main(String[] args) throws Exception {
-        KeyManagerFactory kmf;
-        X509KeyManager km;
-        char [] password = {' '};
 
-        // check for bug 6302126
+    private static KeyManagerFactory kmf;
+    private static X509KeyManager km;
+
+    @BeforeAll
+    public static void beforeAll() throws Exception {
         kmf = KeyManagerFactory.getInstance("NewSunX509");
-        kmf.init((KeyStore)null, password);
 
-        // check for 6302321
+        // creating a new keystore
+        final SecureRandom secureRandom = new SecureRandom();
+        final KeyPair keyPair = generateRSAKeyPair(secureRandom);
+        final X509Certificate originServerCert =
+                generateCert(keyPair, secureRandom, "subject");
+        final KeyStore ks = generateKeyStore(keyPair.getPrivate(),
+                new Certificate[]{originServerCert});
+
+        kmf.init(ks, null);
         km = (X509KeyManager) kmf.getKeyManagers()[0];
-        X509Certificate[] certs = km.getCertificateChain("doesnotexist");
-        PrivateKey priv = km.getPrivateKey("doesnotexist");
-        if (certs != null || priv != null) {
-            throw new Exception("Should return null if the alias can't be found");
-        }
+    }
+
+    private X509KeyManager generateNullKm() throws Exception {
+        char[] password = {' '};
+        kmf.init((KeyStore) null, password);
+        return (X509KeyManager) kmf.getKeyManagers()[0];
+    }
+
+    @Test
+    public void JDK6302126Test() throws Exception {
+        // check for bug 6302126
+
+        generateNullKm();
+    }
+
+    @Test
+    public void JDK6302304Test() throws Exception {
+        // check for bug 6302304
+
+        final X509KeyManager km = generateNullKm();
 
-        // check for 6302271
-        String[] clis = km.getClientAliases("doesnotexist", null);
-        if (clis != null && clis.length == 0) {
-            throw new Exception("Should return null instead of empty array");
-        }
-        String[] srvs = km.getServerAliases("doesnotexist", null);
-        if (srvs != null && srvs.length == 0) {
-            throw new Exception("Should return null instead of empty array");
-        }
-
-        // check for 6302304
         km.getServerAliases(null, null);
```

Comment:
I know that these are not part of this change, but shouldn't we be testing that all these are indeed returning null?